File: examples/rest_orders.js
import { PredictionMarketsAgentClient } from "../dist/agent-client.js";

const BASE_URL = process.env.PM_BASE_URL || "http://127.0.0.1:3002/prediction-markets";
const API_KEY = process.env.PM_AGENT_API_KEY || "";
const ORDER_SIDE = process.env.PM_ORDER_SIDE || "buy_yes";
const ORDER_PRICE_TENTHS = Number(process.env.PM_ORDER_PRICE_TENTHS || 543);
const ORDER_SIZE_SHARES = Number(process.env.PM_ORDER_SIZE_SHARES || 1);
const CANCEL_AFTER_PLACE = process.env.PM_CANCEL_AFTER_PLACE !== "0";

if (!API_KEY) {
  console.error("PM_AGENT_API_KEY is required.");
  process.exit(1);
}

function logEvent(label, payload) {
  console.log(`\n[${label}]`);
  console.log(JSON.stringify(payload, null, 2));
}

async function main() {
  const client = new PredictionMarketsAgentClient({
    baseUrl: BASE_URL,
    apiKey: API_KEY,
  });

  const [me, summary, currentMarket] = await Promise.all([
    client.getMe(),
    client.getAccountSummary(),
    client.getCurrentMarket(),
  ]);
  logEvent("me", me);
  logEvent("account summary", summary);
  logEvent("current market", currentMarket);

  if (!currentMarket.market_id) {
    throw new Error("No active market.");
  }

  const placed = await client.placeOrderRest({
    market_id: currentMarket.market_id,
    side: ORDER_SIDE,
    price_tenths: ORDER_PRICE_TENTHS,
    size_shares: ORDER_SIZE_SHARES,
    client_order_id: `rest-js-${Date.now()}`,
  });
  logEvent("placed", placed);

  const fetched = await client.getOrder(placed.order.order_hash);
  logEvent("fetched", fetched);

  if (CANCEL_AFTER_PLACE) {
    const cancelled = await client.cancelOrderRest(placed.order.order_hash);
    logEvent("cancelled", cancelled);
  }
}

main().catch(error => {
  console.error(error);
  process.exit(1);
});

File: examples/rest_orders.py
import json
import os
import sys

from src.agent_client import PredictionMarketsAgentClient


BASE_URL = os.environ.get("PM_BASE_URL", "http://127.0.0.1:3002/prediction-markets")
API_KEY = os.environ.get("PM_AGENT_API_KEY", "")
ORDER_SIDE = os.environ.get("PM_ORDER_SIDE", "buy_yes")
ORDER_PRICE_TENTHS = int(os.environ.get("PM_ORDER_PRICE_TENTHS", "543"))
ORDER_SIZE_SHARES = float(os.environ.get("PM_ORDER_SIZE_SHARES", "1"))
CANCEL_AFTER_PLACE = os.environ.get("PM_CANCEL_AFTER_PLACE", "1") != "0"


def log_event(label, payload):
    print(f"\n[{label}]")
    print(json.dumps(payload, indent=2))


def main():
    if not API_KEY:
        print("PM_AGENT_API_KEY is required.", file=sys.stderr)
        sys.exit(1)

    client = PredictionMarketsAgentClient(BASE_URL, API_KEY)
    me = client.get_me()
    summary = client.get_account_summary()
    current_market = client.get_current_market()
    log_event("me", me)
    log_event("account summary", summary)
    log_event("current market", current_market)

    market_id = current_market.get("market_id")
    if not market_id:
        raise RuntimeError("No active market.")

    placed = client.place_order_rest(
        {
            "market_id": market_id,
            "side": ORDER_SIDE,
            "price_tenths": ORDER_PRICE_TENTHS,
            "size_shares": ORDER_SIZE_SHARES,
            "client_order_id": f"rest-py-{int(__import__('time').time() * 1000)}",
        }
    )
    log_event("placed", placed)

    fetched = client.get_order(placed["order"]["order_hash"])
    log_event("fetched", fetched)

    if CANCEL_AFTER_PLACE:
      cancelled = client.cancel_order_rest(placed["order"]["order_hash"])
      log_event("cancelled", cancelled)


if __name__ == "__main__":
    main()
